dbt/adapters/factory.py

```diff
@@ -188,7 +188,7 @@ def get_include_paths(self, name: Optional[str]) -> List[Path]:
     def get_adapter_type_names(self, name: Optional[str]) -> List[str]:
         return [p.adapter.type() for p in self.get_adapter_plugins(name)]
 
-    def get_adapter_constraint_support(self, name: Optional[str]) -> List[str]:
+    def get_adapter_constraint_support(self, name: Optional[str]) -> Dict[str, str]:
```

Comment:
Do we still need the `# type: ignore`? My guess is that we needed that precisely because this type annotation was incorrect.

Reply:
Removed in https://github.com/dbt-labs/dbt-adapters/pull/254/commits/734f141d659b698f12997db4ab2572b3d1ead9e7

Reply:
Here's the result:

```
dbt/adapters/factory.py:192: error: "AdapterProtocol[Any, Any, Any, Any]" has
no attribute "CONSTRAINT_SUPPORT"  [attr-defined]
            return self.lookup_adapter(name).CONSTRAINT_SUPPORT
                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
dbt/adapters/factory.py:192: error: Argument 1 to "lookup_adapter" of
"AdapterContainer" has incompatible type "str | None"; expected "str" 
[arg-type]
            return self.lookup_adapter(name).CONSTRAINT_SUPPORT
                                       ^~~~
Found 2 errors in 1 file (checked 48 source files)
```

Reply:
Yeah we can leave it for now I think this is because the method's name argument is `Optional`